### ISSUE TITLE
Inline environment variables are not set

### ISSUE THREAD

**Issue by NullVoxPopuli** [stackblitz/webcontainer-core#164]
**Describe the bug:**
Probably closely related to: https://github.com/stackblitz/webcontainer-core/issues/163

**Link to the blitz that caused the error:** n/a

**Steps to reproduce:**

1. Run node code that throws an exception, [such as here ](https://stackblitz.com/github/ember-cli/ember-new-output/tree/stable?file=package.json)
2. run: `NODE_OPTIONS='--trace-warnings --unhandled-rejections=strict' npm start`
3. observe that `--trace-warnings` and `--unhandled-rejections=strict` flags are not respected

**Expected behavior:**

the environment variable `NODE_OPTIONS` is set for the duration of the command

**Desktop (please complete the following information):**

   - OS: Windows 10
   - Browser: Chrome
   - Version: 91.0.4472.77 (Official Build) (64-bit)

**Comment by SamVerschueren** [comment]
Hi @NullVoxPopuli, thanks for reporting any issues you run into. 

Inline variables are supported actually. Try running `HELLO=world env` or try reading it from `process.env` in a Node.js script, it will work.

However, `npm` is not the npm client from local, that would work very badly in the browser. Basically it's a proxy to our own `turbo` client, so it doesn't have all the features (yet).

Can you explain your use cases or what you're trying to do here?

**Comment by NullVoxPopuli** [comment]
>  is not the npm client from local, that would work very badly in the browser. Basically it's a proxy to our own turbo 

ah ok, that makes sense, thanks!

> Can you explain your use cases or what you're trying to do here?

With newer node versions, I was just trying to get rid of these warnings:
```
(node:6) UnhandledPromiseRejectionWarning: TypeError: Failed to fetch
(Use `node --trace-warnings ...` to show where the warning was created)
(node:6) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:6) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

Locally, I have this in my `.bash_profile`:
```
NODE_OPTIONS='--trace-warnings --unhandled-rejections=strict'
```

**Comment by SamVerschueren** [comment]
Chiming in @d3lm to see if this is something we could/should add? Guess it's quite niche as long as you can't create your own personal shell profile. Which is definitely something we want over time.

**Comment by d3lm** [comment]
I agree that having a custom shell profile where you can add aliases, define env variables etc. is a nice to have and is will very likely be supported at some point.

For the node flags, that's a good point. Right now not all flags are supported and I have created a ticket for this. Thanks for letting us know.

**Comment by d3lm** [comment]
Hey 👋 So `--unhandled-rejections=strict` should be supported, but `--trace-warnings` is not yet supported. And I can confirm that we currently don't support a custom shell profile. I can't speak to an exact timeline but we'll keep you posted here if there's any progress on any of the issues mentioned.
